test/source-maps-test.ts

```diff
@@ -1,27 +1,48 @@
 import * as assert from "assert";
+import * as fs from "fs";
+import * as path from "path";
 
 import {transform} from "../src";
 
 describe("source maps", () => {
-  it("generates a simple line-based source map", () => {
-    const result = transform(
-      `\
+  const testCase = (simple: boolean): void => {
+    const source = `\
       import a from "./a";
       const x: number = 1;
       console.log(x + 1);
-    `,
-      {
-        transforms: ["imports", "typescript"],
-        sourceMapOptions: {compiledFilename: "test.js"},
-        filePath: "test.ts",
-      },
-    );
+    `;
+    const result = transform(source, {
+      transforms: ["imports", "typescript"],
+      sourceMapOptions: {compiledFilename: "test.js", simple},
+      filePath: "test.ts",
+    });
+    const simpleMappings = "AAAA;AACA;AACA;AACA";
     assert.deepEqual(result.sourceMap, {
       version: 3,
       sources: ["test.ts"],
       names: [],
-      mappings: "AAAA;AACA;AACA;AACA",
+      mappings: simple ? simpleMappings : result.sourceMap!.mappings,
       file: "test.js",
     });
+    if (!simple) {
+      if (process.env.WRITE_SOURCE_MAPS) {
```

Comment:
How have you been using this, out of curiosity? It might be something that could be added to the demo website instead (though certainly that would be out of scope for this PR), or it might be less necessary if you hard-code the expected mappings.

Reply:
I've been using that with source map visualization tools - specifically https://evanw.github.io/source-map-visualization/ and https://sokra.github.io/source-map-visualization/ (via https://www.npmjs.com/package/source-map-visualize) - to validate that the mappings were correct.